### ISSUE TITLE
Plan on youtube music

### ISSUE THREAD

**Issue by sJJdGG** [simon-weber/gmusicapi#609]
As you may have heard google has released a new service called youtube music which has lots of similarity with play music. There are news that google eventually plans to replace play music with youtube music.

The good thing about youtube music right now is less authorization headache and new information like destinction between singles and albums.

would be really good this library supported it.

Just as a recommendation and knowing everyone's opinion.

**Comment by thebigmunch** [comment]
They have also said they don't plan on replacing Google Music until they've implemented all the functionality into YouTube Music.

As far as implementing it in this library, I'd definitely say it should be in a separate library. You don't just slap in a completely different service into a library because it's *technically* a replacement for it.

Also, the mobile APIs are all done using protocol buffers. Adds an extra pain to using the mobile APIs, especially since you need to be using much older versions of Android/iOS to actually capture these calls now.

So, for someone implementing a YouTube Music library, they're left with using the HTTP API... oh wait, there isn't one, because Google. So, you're left in the same boat as the Webclient in gmusicapi. It can be a giant pain to keep it functioning. Also, this kinda goes against it being less of an authorization headache.

**Comment by simon-weber** [comment]
Yup, I don't have any immediate plans to look at youtube music. I suspect I will eventually so I can keep [autoplaylists](https://github.com/simon-weber/Autoplaylists-for-Google-Music) running, though by the time I get to that someone else may have already figured out the few calls I need.

I agree that it should probably be a different library. One case I'd consider adding support directly to gmusicapi would be if it turned out that the apis were very similar.

**Comment by thebigmunch** [comment]
> I agree that it should probably be a different library. One case I'd consider adding support directly to gmusicapi would be if it turned out that the apis were very similar.

They're aren't terribly different to Google Music web calls at this point, though I consider that a negative. Though you do have to sift through a lot of cruft to figure out what's necessary and not.

Currently there's almost nothing to the functionality and less than a handful of calls.

**Comment by thebigmunch** [comment]
PS

As I've already told a few people who have asked me if I would make a YouTube Music library, I'd have already made one if Google provided a proper web API like [Spotify](https://developer.spotify.com/documentation/web-api/) provides. Dunno if I'd make one before YouTube Music supported uploading your own songs, but I can't say I wouldn't for sure ^_^

**Comment by sigma67** [comment]
I did a little research into the current state of YT Music API access.

While there is no dedicated API for YT Music currently, there is the YouTube Data API v3, which works to create playlists for YouTube Music. You can also search for entries from YouTube Music using the standard YouTube search by adding the term "Auto-generated by YouTube" to the search, which is present in all music-only tracks (with album cover art) generated by YouTube. So while it is a little hacky, playlist management is definitely possible right now

**Comment by sigma67** [comment]
I made a thing for YouTube Music: https://github.com/sigma67/ytmusicapi/blob/master/ytmusicapi/ytmusic.py

It's not pretty, but it works for my purposes and doesn't hit the quota limits

**Comment by Suncatcher** [comment]
> They have also said they don't plan on replacing Google Music until they've implemented all the functionality into YouTube Music.

This is no longer the case. Google [officially announced](https://youtube.googleblog.com/2020/05/youtube-music-transfer-google-play-music-library.html) it will shutdown GMusic by the end of the year. So it probably not worth investing more efforts into development of this API.
